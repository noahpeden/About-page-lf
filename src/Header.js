import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
const Wrapper = styled.div`

`

export default (props) => (
  <div>
    <Link to="/history">Past Searches</Link>
    <a onClick={props.getRandom}>Random City</a>
  </div>
)