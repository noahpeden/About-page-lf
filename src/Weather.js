import React from 'react'

export default () => (
  <div>
    This is the weather you searched.
  </div>
)