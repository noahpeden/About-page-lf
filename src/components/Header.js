import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';


const Wrapper = styled.div`
	margin-top: 50px;
	margin-bottom: 40%;
	color: white;
	height: 50px;
	width: 100%;
	display: flex;
	justify-content: space-around;
	a {
		cursor: pointer;
	}
	a:hover {
		text-decoration: underline;
	}
	div:hover {
		text-decoration: underline;
	}
`;

export default props =>
	<Wrapper>
		<Link style={{ textDecoration: 'none', color: 'white' }} to="/searches">
			<div>PAST SEARCHES</div>
		</Link>
		<a onClick={props.getRandom}>RANDOM</a>
		<a onClick={props.getCurrent}>YOUR LOCATION</a>
	</Wrapper>
