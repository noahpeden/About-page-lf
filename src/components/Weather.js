import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
	ul {
		text-align: center;
		margin-top: 20px;
		font-size: 20px;
	}
`;

export default props =>
	<div>
		<Wrapper>
			<ul>
				<li>
					{props.currently}
				</li>
				<li>
					{props.description}
				</li>
				<li>
					{props.cityName}
				</li>
				<li>
					{props.timeStamp}
				</li>
			</ul>
		</Wrapper>
	</div>;
