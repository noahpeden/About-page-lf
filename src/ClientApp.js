import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import fire from './firebase';
import { Link } from 'react-router-dom';
import Header from './Header';
import Searches from './Searches';
import Weather from './Weather';
import randomCities from './randomCities';
import styled from 'styled-components';

class ClientApp extends Component {
	constructor(props) {
		super(props);
		this.state = {
			searches: [],
			name: '',
			city: '',
			timeStamp: ''
		};
		this.addSearch = this.addSearch.bind(this);
		this.getWeather = this.getWeather.bind(this);
		this.handleChange = this.handleChange.bind(this);
		this.getRandom = this.getRandom.bind(this);
	}

	componentWillMount() {
		let searchesRef = fire.database().ref('searches').orderByKey();
		searchesRef.on('child_added', snapshot => {
			let search = { text: snapshot.val(), id: snapshot.key };
			this.setState({ searches: [search].concat(this.state.searches) });
		});
	}

	addSearch(e) {
		this.getWeather(this.state.name);
		e.preventDefault();
		fire.database().ref('searches').push(this.inputEl.value);
		this.inputEl.value = '';
	}

	getWeather(city) {
		let date = new Date();
		let str = `${date.getFullYear()}-${date.getMonth() +
			1}-${date.getDate()} ${date.getHours()}:${date.getMinutes()}:${date.getSeconds()}`;
		const APIKey = '2eacf5cd8e08d4adc524186577921400';
		const URL = `http://api.openweathermap.org/data/2.5/weather?q=${city}&APPID=${APIKey}&units=imperial`;
		fetch(URL)
			.then(response => {
				return response.json();
			})
			.then(response => {
				this.setState({
					currently: response.main.temp,
					description: response.weather[0].description,
					cityName: response.name,
					timeStamp: str
				});
			});
	}

	getRandom() {
		let city = randomCities[Math.floor(Math.random() * randomCities.length)];
		this.getWeather(city);
	}

	handleChange(event) {
		this.setState({
			name: event.target.value
		});
	}

	render() {
		return (
			<div className="App">
				<Header getRandom={this.getRandom} />
				<form onSubmit={this.addSearch}>
					<input
						type="text"
						onChange={this.handleChange}
						placeholder="Search a city for the current weather"
						ref={el => (this.inputEl = el)}
					/>
					<input type="submit" />
				</form>
				<Weather
					currently={this.state.currently}
					description={this.state.description}
					cityName={this.state.cityName}
					timeStamp={this.state.timeStamp}
				/>
			</div>
		);
	}
}

export default ClientApp;
