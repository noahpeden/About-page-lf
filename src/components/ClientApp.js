import React, { Component } from 'react';
import fire from '../firebase';
import Header from './Header';
import Weather from './Weather';
import randomCities from '../assets/randomCities';
import styled from 'styled-components';

class ClientApp extends Component {
	state = {
		searches: [],
		name: '',
		city: '',
		currently: '',
		description: '',
		timeStamp: ''
	};

	addSearch = e => {
		if (e.key === 'Enter') {
			this.getWeather(this.inputEl.value);
			this.inputEl.value = '';
		}
	};

	getWeather = city => {
		const APIKey = '2eacf5cd8e08d4adc524186577921400';
		const URL = `http://api.openweathermap.org/data/2.5/weather?q=${city}&APPID=${APIKey}&units=imperial`;
		this.fetchData(URL);
	};

	fetchData = URL => {
		let date = new Date();
		let str = `${date.getFullYear()}-${date.getMonth() +
			1}-${date.getDate()} ${date.getHours()}:${date.getMinutes()}:${date.getSeconds()}`;
		fetch(URL)
			.then(response => {
				return response.json();
			})
			.then(response => {
				this.setState({
					currently: `Currently: ${response.main.temp}`,
					description: `Description: ${response.weather[0].description}`,
					cityName: `Location: ${response.name}`,
					timeStamp: str
				});
			})
			.then(response => {
				const search = {
					cityName: this.state.cityName,
					timeStamp: this.state.timeStamp,
					description: this.state.description,
					currently: this.state.currently
				};
				fire.database().ref('search').push(search);
			});
	};

	getRandom = () => {
		let city = randomCities[Math.floor(Math.random() * randomCities.length)];
		this.getWeather(city);
	};

	getCurrentLocation = () => {
		const APIKey = '2eacf5cd8e08d4adc524186577921400';
		navigator.geolocation.getCurrentPosition(position => {
			const URL = `http://api.openweathermap.org/data/2.5/weather?lat=${position
				.coords.latitude}&lon=${position.coords
				.longitude}&APPID=${APIKey}&units=imperial`;
			this.fetchData(URL);
		});
	};

	render() {
		const Wrapper = styled.div`
			display: flex;
			justify-content: center;
			align-items: center;
			color: white;
			h1 {
				color: white;
				font-size: 40px;
			}
			input,
			select,
			textarea {
				color: white;
			}
			input {
				width: 315px;
				margin-left: 50px;
				margin-top: 50px;
				font-size: 20px;
				outline: none;
				border: 0px solid white;
				border-bottom-width: 1px;
				background-color: transparent;
			}
			button {
				background: none;
				border: none;
				border-radius: 4%;
				outline: none;
				color: white;
				font-size: 20px;
			}
			button:hover {
				background: white;
				color: black;
				cursor: pointer;
			}
			input::placeholder {
				color: white;
			}
			input:focus::-webkit-input-placeholder {
				color: transparent;
			}
			input:focus:-moz-placeholder {
				color: transparent;
			}
		`;
		return (
			<Wrapper>
				<div className="App">
					<Header
						getRandom={this.getRandom}
						getCurrent={this.getCurrentLocation}
					/>
					<h1>DAGON WEATHER APP</h1>
					<input
						onKeyPress={this.addSearch}
						type="text"
						onChange={this.handleChange}
						placeholder="Search a city for the current weather"
						ref={el => (this.inputEl = el)}
					/>
					<Weather
						currently={this.state.currently}
						description={this.state.description}
						cityName={this.state.cityName}
						timeStamp={this.state.timeStamp}
					/>
				</div>
			</Wrapper>
		);
	}
}

export default ClientApp;
