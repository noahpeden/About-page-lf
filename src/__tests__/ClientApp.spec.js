import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import ClientApp from '../components/ClientApp';
import Weather from '../components/Weather'
import Header from '../components/Header'

test('ClientApp renders correctly', () => {
	const component = shallow(<ClientApp />);
	expect(component).toMatchSnapshot();
});

test('Should render weather', () => {
  const component = shallow(<ClientApp/>)
  expect(component.find(Weather).length).toEqual(1)
})

test('Should render header', () => {
  const component = shallow(<ClientApp/>)
  expect(component.find(Header).length).toEqual(1)
})

