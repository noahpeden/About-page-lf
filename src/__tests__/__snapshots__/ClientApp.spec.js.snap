// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientApp renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <div
        className="App"
    >
        <Unknown
            getCurrent={[Function]}
            getRandom={[Function]}
        />
        <h1>
            DAGON WEATHER APP
        </h1>
        <input
            onChange={undefined}
            onKeyPress={[Function]}
            placeholder="Search a city for the current weather"
            type="text"
        />
        <br />
        <Unknown
            cityName={undefined}
            currently=""
            description=""
            timeStamp=""
        />
    </div>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <div
            className="App"
      >
            <Unknown
                  getCurrent={[Function]}
                  getRandom={[Function]}
            />
            <h1>
                  DAGON WEATHER APP
            </h1>
            <input
                  onChange={undefined}
                  onKeyPress={[Function]}
                  placeholder="Search a city for the current weather"
                  type="text"
            />
            <br />
            <Unknown
                  cityName={undefined}
                  currently=""
                  description=""
                  timeStamp=""
            />
      </div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ClientApp />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ClientApp {
        "_reactInternalInstance": [Circular],
        "addSearch": [Function],
        "context": Object {},
        "fetchData": [Function],
        "getCurrentLocation": [Function],
        "getRandom": [Function],
        "getWeather": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "city": "",
          "currently": "",
          "description": "",
          "loading": false,
          "name": "",
          "searches": Array [],
          "timeStamp": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <div
                    className="App"
          >
                    <Unknown
                              getCurrent={[Function]}
                              getRandom={[Function]}
                    />
                    <h1>
                              DAGON WEATHER APP
                    </h1>
                    <input
                              onChange={undefined}
                              onKeyPress={[Function]}
                              placeholder="Search a city for the current weather"
                              type="text"
                    />
                    <br />
                    <Unknown
                              cityName={undefined}
                              currently=""
                              description=""
                              timeStamp=""
                    />
          </div>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <div
                    className="App"
          >
                    <Unknown
                              getCurrent={[Function]}
                              getRandom={[Function]}
                    />
                    <h1>
                              DAGON WEATHER APP
                    </h1>
                    <input
                              onChange={undefined}
                              onKeyPress={[Function]}
                              placeholder="Search a city for the current weather"
                              type="text"
                    />
                    <br />
                    <Unknown
                              cityName={undefined}
                              currently=""
                              description=""
                              timeStamp=""
                    />
          </div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ClientApp />,
}
`;
